### ISSUE TITLE
How the number of identified proteins is calculated in report.stats ?

### ISSUE THREAD

**Issue by KR-15** [vdemichev/DiaNN#180]
Hi,

Thank you for your amazing work done !

I would like to know how the number of identified proteins is calculated in the report.stats ?

I am asking this question because I never find the same number of identified proteins in the report.pg_matrix.

Do you apply some filters ? I read in the log that for protein inference, inference is performed using proteotypic peptides only but when I applied this filter in the global report, I do not get the number of identified proteins which is displayed in the report.stats.

Thank you for your response.

Kind regards

Kevin

**Comment by vdemichev** [comment]
Hi Kevin,

You can reproduce it using the main report obtained with Precursor FDR set to 100%. For this, filter based on Proteotypic == 1 &  Protein.Q.Value <= 0.01.

Best,
Vadim

**Comment by KR-15** [comment]
Hi Vadim,

Thank you for your answer. I have tried what you suggest but it did not work for me. 

I work on the main report.tsv file and I have applied both filters that you suggest.  Then, I took the "Protein.Group" list and removed duplicate groups and did not get the same number mentionned in the "report.stats.tsv" file. 

In addition, why do I get also a different number of 'Protein.Group' in the "report.pg_matrix.tsv" ? 

I can send you the "report.stats.tsv", "report.tsv" and the "report.pg_matrix.tsv"

Even when I filter with Global.Q.value <= 0.01 as mentionned in the "Output" part of the documentation, I don't have the same number of protein.groups that I get in the "report.pg_matrix.tsv"

[report_github.zip](https://github.com/vdemichev/DiaNN/files/7170865/report_github.zip)

Could you summarize what are the filtering steps that are applied from the main report in order to get the number of protein.ids mentionned in the report.stats and also the filtering steps to get the final "report.pg_matrix.tsv" please ?

Thank you.

Best,

Kevin

**Comment by vdemichev** [comment]
`
> library(diann)
> pg <- diann_load('C:/Temp/report_github/report.pg_matrix.tsv')
> dim(pg)
[1] 9207    6
> length(unique(pg$Protein.Group))
[1] 9207
> df <- diann_load('C:/Temp/report_github/report.tsv')
> length(unique(df$Protein.Group[df$Global.Q.Value <= 0.01 & df$Global.PG.Q.Value <= 0.01]))
[1] 9207
> length(unique(df$Genes[df$Proteotypic == 1 & df$Protein.Q.Value <= 0.01]))
[1] 7648
> stats <- diann_load('C:/Temp/report_github/report.stats.tsv')
> stats$Proteins.Identified
[1] 7653
> range(df$Q.Value)
[1] 0.000000001 0.009989740
`
As you can see, the only discrepancy is 7648 vs 7653 unique proteins. This is because the main report was filtered at 1% precursor FDR. If you filter at 50% precursor FDR there will be no discrepancy.

**Comment by KR-15** [comment]
Hi Vadim,

Finally, you get the same number working with the "Genes" column instead of "Protein.Group" column after keeping unique values. It is clearer for me now.

Thanks a lot !

Kevin
